lsp-dart-utils.el

```diff
@@ -118,6 +118,14 @@ Order is important."
       (append list (list value))
     list))
 
+(defun lsp-dart-flutter-snap-install-p ()
+  ;; Detecting whether this is a Linux system with a Snap style install
+  (and (string= system-type "gnu/linux")
```

Comment:
Can' t snap be used on other OSs like apple?

Reply:
I honestly don't know. I restricted the function as much as possible as not to cause problems for others. I'll look into it.

Reply:
So reading the snap installer docs[1] and the flutter install docs[2], I can't seem to find any references to anything but Linux regarding snaps. The only references I could find for macs is the build tools[4] and the code also makes a lot of references to Linux specific things[3] which would be hard to make run in any other system. 

I guess you can make a setup using Virtual Machines (or WSL2) but should this functionality cover such a scenario?

[1] https://snapcraft.io/docs/installing-snapd 
[2] https://docs.flutter.dev/get-started/install
[3] https://github.com/snapcore/snapd
[4] https://snapcraft.io/docs/release-notes-snapcraft-3-0

Reply:
No problem, I'm ok with that for now